src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = Object.assign({}, state);
+  const result = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(cloneState, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const arrElem of obj.keysToRemove) {
```

Comment:
You can use desctructuring to get all the needed props of action (type, keysToRemove, extraData...)